escodegen.js

```diff
@@ -1145,7 +1145,17 @@
                 result = join(result, this.generateExpression(stmt.id, Precedence.Sequence, E_TTT));
             }
             if (stmt.superClass) {
-                fragment = join('extends', this.generateExpression(stmt.superClass, Precedence.Assignment, E_TTT));
+                var superClassExpression = this.generateExpression(stmt.superClass, Precedence.Assignment, E_TTT);
```

Comment:
None of the rest of this is necessary. [ClassHeritage](https://tc39.es/ecma262/#prod-ClassHeritage) is a LeftHandSideExpression, not an AssignmentExpression. We should just replace `Precedence.Assignment` on this line with `Precedence.Unary`.

Reply:
Will try

Reply:
It works, thank you